service/azureconfig/v1/key/types.go

```diff
@@ -0,0 +1,39 @@
+package key
+
+import "fmt"
+
+type Azure struct {
+	HostCluster AzureHostCluster
+	Location    string
+}
```

Comment:
For a lack of better idea I added a pure data type, which provides Validate method, so it's reused in `New`. I'm going to include information required to create a new client as well. So we don't pass client.AzureConfig around.

Reply:
I like the approach and the use of the Validate method.

Reply:
I would have created a new package for this. This is not about the things the key package is usually concerned about.

Reply:
Bad thing about this place is that `v1/key` is imported in the main service. This thing must be resource version independent like the CR is. I would put that into `azureconfig/types.go`, but that will create cyclic dependency. I'm going to create a `azureconfig/config/types.go` and put it there.

Thoughts?

Reply:
Why not keeping it in this repo in `service/config/` or something?

Reply:
I meant this repo. `/service/azureconfig/config`